docs/data/toolpad/concepts/data-providers.md

```diff
@@ -0,0 +1,117 @@
+# Data Providers
+
+<p class="description">Bring tabular data to the frontend with server-side pagination and filtering.</p>
+
+Toolpad functions are great to bring some backend state to the page, but they fall short when it comes to offering pagination and filtering capabilities from the server. Toolpad offers a special construct to enable this use case: Data providers. Data providers abstract server-side collections. They could be database tables, or REST APIs, or any data that represents a set of records that share a common interface. Data providers are defined as server-side objects and can be directly connected to a data grid to make it fully interactive.
```

Comment:
Maybe this can be split into 2 paragraphs, seems like a large chunk of text to read right at the start. :)

Reply:
Most writing style guides I could find (e.g. https://libguides.hull.ac.uk/writing/paras) suggest between 100-200 words per paragraph or less than 6 sentences.

Reply:
Got it, it's fine to keep it too! Just personal opinion, and website users are particularly lazy readers as opposed to people who read studies, books or newspapers :D